src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectLogsActionTests.cs

```diff
@@ -239,27 +253,41 @@ public Task LogsWildcardActionTest(TargetFrameworkMoniker tfm, LogFormat logForm
 
                     using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(CommonTestTimeouts.LogsTimeout);
 
-                    CollectionRuleActionResult result;
                     try
                     {
                         await action.StartAsync(cancellationTokenSource.Token);
 
                         await runner.SendCommandAsync(TestAppScenarios.Logger.Commands.StartLogging);
 
-                        result = await action.WaitForCompletionAsync(cancellationTokenSource.Token);
+                        using Stream readerStream = logEntriesPipe.Reader.AsStream();
+
+                        await LogsTestUtilities.ValidateLogsEquality(readerStream, callback, logFormat, _outputHelper);
+
+                        // Note: Do not wait for completion of the action execution. No more relevant data will be produced;
+                        // the code would only be waiting for the action to end. Ideally the action is gracefully stopped at
+                        // this point.
                     }
                     finally
                     {
                         await Tools.Monitor.DisposableHelper.DisposeAsync(action);
                     }
-
-                    string egressPath = ActionTestsHelper.ValidateEgressPath(result);
-
-                    using FileStream logsStream = new(egressPath, FileMode.Open, FileAccess.Read);
-                    Assert.NotNull(logsStream);
-
-                    await LogsTestUtilities.ValidateLogsEquality(logsStream, callback, logFormat, _outputHelper);
                 });
+            },
+            services =>
+            {
+                services.RegisterProvider<PipeEgressOptions, PipeEgressProvider>(PipeEgressProvider.Name);
```

Comment:
After egress extensibility, will this still be an inproc implementation? It seems like we would want to accommodate that option for unit testing.

Reply:
It can also be out-of-proc that forwards its stdin to a Unix domain socket if we want to avoid special casing in-proc implementations.